core/Objects/NPMClient.cs

```diff
@@ -146,6 +171,204 @@ public IPackage[] FindPlugins(Uri[] registries)
+            var tgz = $"{name}-{ctrlr.CmfPackage.Version}.tgz";
+            Log.Debug($"Trying to publish {name} to {this.baseUrl}");
+            JObject root = null;
+            
+            try
+            {
+                Log.Debug("Load package content...");
+                using var fileStream = package.OpenRead();
+                using var memoryStream = new MemoryStream();
+                
+                fileStream.CopyTo(memoryStream);  // this can lead to large memory consumption
+                byte[] fileBytes = memoryStream.ToArray();
+
+                string dataBase64 = Convert.ToBase64String(fileBytes);
+
+                using SHA1 sha1 = SHA1.Create();
+                byte[] sha1HashBytes = sha1.ComputeHash(fileBytes);
+                var sha1Hash = BitConverter.ToString(sha1HashBytes).Replace("-", "").ToLower();
+
+                using SHA512 sha512 = SHA512.Create();
+                byte[] sha512HashBytes = sha512.ComputeHash(fileBytes);
+                var sha512_64 = Convert.ToBase64String(sha512HashBytes);
+                // var sha512Hash = BitConverter.ToString(sha512HashBytes).Replace("-", "").ToLower();
+                var sha512Hash = $"sha512-{sha512_64}";
+                Log.Debug($"Package content with hash {sha512Hash} and checksum {sha1Hash}");
+                
+                root = JObject.Parse(
+                    $$"""
+                    { 
+                        "_id": "{{name}}",
+                        "name": "{{name}}",
+                        "description": "{{ctrlr.CmfPackage.Description}}",
+                        "dist-tags": { "latest": "{{ctrlr.CmfPackage.Version}}" },
+                        "versions": {
+                            "{{ctrlr.CmfPackage.Version}}" : {{manifest}}
+                        },
+                        "access": null,
+                        "_attachments": {
+                          "{{tgz}}": {
+                            "content_type": "application/octet-stream",
+                            "data": "{{dataBase64}}",
+                            "length": {{package.Length}}
+                          }
+                        }
+                    }
+                    """);
+                // patch version manifest
+                root["versions"][ctrlr.CmfPackage.Version]["_id"] = $"{name}@{ctrlr.CmfPackage.Version}";
+                root["versions"][ctrlr.CmfPackage.Version]["_cliVersion"] = ExecutionContext.CurrentVersion;
+                root["versions"][ctrlr.CmfPackage.Version]["_integrity"] = sha512Hash;
+                root["versions"][ctrlr.CmfPackage.Version]["dist"] = JObject.Parse($$"""
+                      {
+                        "integrity": "{{sha512Hash}}",
+                        "shasum": "{{sha1Hash}}",
+                        "tarball": "{{this.baseUrl.Replace("https://", "http://")}}/{{name}}/-/{{tgz}}"
```

Comment:
Not familiar with how NPM handles the manifests internally, so sorry if this is a dumb question, but why do we need to convert from HTTPS to HTTP here?